marshmallow_sqlalchemy/schema.py

```diff
@@ -111,21 +112,52 @@ class Meta:
         schema = UserSchema()
 
         user = schema.load({'name': 'Bill'}, session=session)
+        existing_user = schema.load({'name': 'Bill'}, instance=User.query.first())
+
+    :param session: Optional SQLAlchemy session; may be overridden in `load.`
+    :param instance: Optional existing instance to modify; may be overridden in `load`.
     """
     OPTIONS_CLASS = ModelSchemaOpts
 
     def __init__(self, *args, **kwargs):
         session = kwargs.pop('session', None)
+        self.instance = kwargs.pop('instance', None)
         super(ModelSchema, self).__init__(*args, **kwargs)
         self.session = session or self.opts.sqla_session
 
+    def get_instance(self, data):
+        """Retrieve an existing record by primary key."""
+        primary_column = get_primary_column(self.opts.model)
+        if primary_column.key in data:
+            return self.session.query(
+                self.opts.model
+            ).filter(
+                primary_column == data[primary_column.key]
+            ).first()
+        return None
+
     @ma.post_load
     def make_instance(self, data):
-        """Deserialize to an instance of the model."""
+        """Deserialize data to an instance of the model. Update an existing row
+        if specified in `self.instance` or loaded by primary key in the data;
+        else create a new row.
+
+        :param data: Data to deserialize.
+        """
+        instance = self.instance or self.get_instance(data)
+        if instance is not None:
+            instance.__dict__.update(data)
```

Comment:
Is this the preferred way to update a SQLA model instance? Using `__dict__.update` rather than `setattr` means we would be bypassing the descriptor protocol, which may or may not have unwanted consequences.